package.json

```diff
@@ -57,12 +59,26 @@
   },
   "devDependencies": {
     "coveralls": "~3.0.2",
+    "husky": "^1.1.0",
     "istanbul": "~0.4.5",
     "jshint": "~2.9.6",
+    "lint-staged": "^7.3.0",
     "mocha": "5.2.0",
     "nock": "10.0.1",
+    "prettier": "^1.14.2",
     "should": "13.2.3",
     "timekeeper": "2.1.2",
     "watch": "~1.0.2"
+  },
+  "husky": {
```

Comment:
How this exactly works? I mean:

* Does some configuration needs to be do on the "Settings / Webhook" sections in the repository to have this working?
* How the "autoformated on commit" works? Adding an additional commit automatically with the fix?

Reply:
Kind of, Husky builds a local precommit hook for you _on your local instance_ - there is no need to add anything extra to the repository on GitHub. It means that the code gets properly formatted prior commit.

see: 
* https://davidwalsh.name/prevent-bad-commits-husky
* https://prettier.io/docs/en/precommit.html

Reply:
Thanks for the links! I didn't know about pre-commit git hooks. Seems cool :)

NTC